### PR TITLE
gen-bundle: Add options to create a bundle from local directory

### PR DESCRIPTION
**Pull request by irori**
@nyaxt Do you think this is worth merging?

### CODE REVIEW

File: go/bundle/cmd/gen-bundle/fromdir.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	if _, err := b.WriteTo(fo); err != nil {
+		return fmt.Errorf("Failed to write exchange. err: %v", err)
+	}
+	return nil
+}
+
+func addEntriesFromDir(dir string, baseURL *url.URL, b *bundle.Bundle) error {
+	files, err := ioutil.ReadDir(dir)
```
[the remark this answers is not in the document]
Comment (hajimehoshi) [reply go/bundle/cmd/gen-bundle/fromdir.go]: or https://golang.org/pkg/path/filepath/#Walk :-)

File: go/bundle/cmd/gen-bundle/fromdir.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	b := &bundle.Bundle{}
+	if err := addEntriesFromDir(dir, parsedBaseURL, b); err != nil {
+		return err
+	}
+	// Move the startURL entry to first.
+	for i, e := range b.Exchanges {
+		if e.Request.URL.String() == parsedStartURL.String() {
+			tmp := b.Exchanges[0]
+			b.Exchanges[0] = e
+			b.Exchanges[i] = tmp
```
[the remark this answers is not in the document]
Comment (irori) [reply go/bundle/cmd/gen-bundle/fromdir.go]: Done.

File: go/bundle/cmd/gen-bundle/fromdir.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (irori) [reply go/bundle/cmd/gen-bundle/fromdir.go]: Done.

File: go/bundle/cmd/gen-bundle/main.go [hunk 3]
```diff
@@ -117,7 +120,16 @@ func run() error {
 
 func main() {
 	flag.Parse()
-	if err := run(); err != nil {
-		log.Fatal(err)
+	if *flagHar != "" {
```
[the remark this answers is not in the document]
Comment (irori) [reply go/bundle/cmd/gen-bundle/main.go]: Done.

File: go/bundle/cmd/gen-bundle/fromdir.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		}
+	}
+
+	if _, err := b.WriteTo(fo); err != nil {
+		return fmt.Errorf("Failed to write exchange. err: %v", err)
+	}
+	return nil
+}
+
+func addEntriesFromDir(dir string, baseURL *url.URL, b *bundle.Bundle) error {
```
[the remark this answers is not in the document]
Comment (irori) [reply go/bundle/cmd/gen-bundle/fromdir.go]: Now it returns exchanges as a return value.

**Comment by irori** [comment]
All comments addressed. PTAL

File: go/bundle/cmd/gen-bundle/main.go [hunk 5]
```diff
@@ -20,7 +20,7 @@ import (
 var (
 	flagHar      = flag.String("har", "", "HTTP Archive (HAR) input file")
 	flagDir      = flag.String("dir", "", "Input directory")
-	flagBaseUrl  = flag.String("baseUrl", "https://example.com/", "Base URL")
+	flagBaseUrl  = flag.String("baseUrl", "", "Base URL")
```
Comment (hajimehoshi) [inline go/bundle/cmd/gen-bundle/main.go]: `flagBaseURL` for Go convention https://stackoverflow.com/questions/38616687/which-way-to-name-a-function-in-go-camelcase-or-semi-camelcase

File: go/bundle/cmd/gen-bundle/main.go [hunk 6]
```diff
@@ -20,7 +20,7 @@ import (
...
 	flagStartUrl = flag.String("startUrl", "", "Entry point URL (relative from -baseUrl)")
```
Comment (hajimehoshi) [inline go/bundle/cmd/gen-bundle/main.go]: ditto

File: go/bundle/cmd/gen-bundle/main.go (hunk 5 above)
Comment (irori) [reply go/bundle/cmd/gen-bundle/main.go]: done

File: go/bundle/cmd/gen-bundle/main.go (hunk 6 above)
Comment (irori) [reply go/bundle/cmd/gen-bundle/main.go]: done

File: go/bundle/cmd/gen-bundle/fromdir.go [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+		return "", fmt.Errorf("Cannot make relative path for %s", path, err)
+	}
+	url, err := baseURL.Parse(filepath.ToSlash(relPath))
+	if err != nil {
+		return "", fmt.Errorf("Failed to construct URL for %s. err: %v", path, err)
+	}
+	return url.String(), nil
+}
+
+// Implements http.ResponseWriter
```
Comment (hajimehoshi) [inline go/bundle/cmd/gen-bundle/fromdir.go]: nit: The comment should start with `responseWriter` for Go convention.

**Comment by irori** [comment]
Thanks for the review!

**Pull request merged**